core/src/test/java/org/elasticsearch/index/query/GeoDataGenerator.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.elasticsearch.common.geo.GeoPoint;
+import org.elasticsearch.common.geo.GeoUtils;
+import org.elasticsearch.test.ElasticsearchTestCase;
+
+/** Helper class for generating geo related data. */
+public class GeoDataGenerator {
```

Comment:
Can we move this to the ```org.elasticsearch.test.geo``` package and look at splitting random point generation methods out of RandomShapeGenerator.java so you can reuse them without requiring s4j and jts which is (LGPL)? At present this class will introduce some code duplication (including yet another BBox class: see issue #10617)